Tribler/Policies/BoostingManager.py

```diff
@@ -0,0 +1,493 @@
+        """
+        if infohash not in self.torrents:
+            return
+
+        self._logger.debug("infohash %s %s %s updated", subject, change_type, hexlify(infohash))
+
+        tdict = self.torrent_db.getTorrent(infohash, keys=['C.torrent_id', 'infohash', 'name',
+                                                           'length', 'category', 'status', 'num_seeders',
+                                                           'num_leechers'])
+
+        if tdict:
+            infohash_str = hexlify(tdict['infohash'])
+
+            new_seed = tdict['num_seeders']
+            new_leecher = tdict['num_leechers']
+
+            if new_seed - self.torrents[tdict['infohash']]['num_seeders'] \
+                    or new_leecher - self.torrents[tdict['infohash']]['num_leechers']:
+                self.torrents[tdict['infohash']]['num_seeders'] = new_seed
+                self.torrents[tdict['infohash']]['num_leechers'] = new_leecher
+                self._logger.info("infohash %s : seeder/leecher changed seed:%d leech:%d",
+                                  infohash_str, new_seed, new_leecher)
+
+    def scrape_trackers(self):
+        """
+        Manually scrape tracker by requesting to tracker manager
+        """
+
+        for infohash in list(self.torrents):
+            # torrent handle
+            lt_torrent = self.session.lm.ltmgr.get_session().find_torrent(lt.big_number(infohash))
+
+            peer_list = []
+            for i in lt_torrent.get_peer_info():
+                peer = LibtorrentDownloadImpl.create_peerlist_data(i)
+                peer_list.append(peer)
+
+            num_seed, num_leech = utilities.translate_peers_into_health(peer_list)
+            self._logger.debug("Seeder/leecher data translated from peers : seeder %s, leecher %s", num_seed, num_leech)
+
+            # check health(seeder/leecher)
+            self.session.lm.torrent_checker.add_gui_request(infohash, True)
+
+    def set_archive(self, source, enable):
+        """
+        setting archive of a particular source. Affect all the torrents in this source
+        """
+        if source in self.boosting_sources:
+            self.boosting_sources[source].archive = enable
+            self._logger.info("Set archive mode for %s to %s", source, enable)
+        else:
+            self._logger.error("Could not set archive mode for unknown source %s", source)
+
+    def start_download(self, torrent):
+        """
+        Start downloading a particular torrent and add it to download list in Tribler
+        """
+        dscfg = DownloadStartupConfig()
+        dscfg.set_dest_dir(self.settings.credit_mining_path)
+        dscfg.set_safe_seeding(False)
+
+        preload = torrent.get('preload', False)
+
+        pstate = CallbackConfigParser()
+        pstate.add_section('state')
+        pstate.set('state', 'engineresumedata', torrent.get('pstate', None))
+
+        # not using Session.start_download because we need to specify pstate
```

Comment:
Possibly something for a next PR but I would rather use `Session.start_download` instead since we now have code to start a download in two places. Is this possible somehow?